### ISSUE TITLE
Improve `computeMedianCV`

### ISSUE THREAD

**Issue by cvanderaa** [UCLouvain-CBIO/scp#7]
The implementation of `computeMedianCV` is messy: 

* Use of long format data that is manipulated with `dplyr` functions
* ~~There is a filter on `SampleType` that is specific for replicating the SCoPE2 analysis. It must be removed for general use~~
* The algorithm for computing the CVs is very convoluted and hard to grasp. This makes unit testing very difficult (see issue #5)
* The computed CV should be stored in the `colData` at the `QFeature` level rather than the `SingleCellExperiment` level.

**Comment by lgatto** [comment]
I'm wondering if it wouldn't be worth considering a *backward compatible* mode if the difference is significant.

**Comment by cvanderaa** [comment]
Sorry, I realized I wrongly used the word *merely*. I meant there is almost no change so in my opinion the current implementation is generic and reproduces the SCoPE2 results accurately enough. However, the implementation still looks convoluted to me. Simplifying the function might lead to significant changes that might require backward compatibility indeed, although I think if it is to only reproduce SCoPE2 results, it might not be worth the effort. We could code the current implementation in the vignette as standalone function.

**Comment by cvanderaa** [comment]
I would think of a simpler implementation like the one in `MSnbase:::featureCV`, but we will have to figure out if this is suitable for single-cell data. Would you have an opinion about it? 

Here are the current implementations of:
- [`scp::computeMedianCV`](https://github.com/UCLouvain-CBIO/scp/blob/45a4768b8ffd9a3ff5c7eed4c1a15af840eb72ad/R/compute_metrics.R#L181-L218)
- [`MSnbase:::featureCV`](https://github.com/lgatto/MSnbase/blob/1082b1dedd97b7e115b4cea256f1e0bd667b2e92/R/functions-MSnSet.R#L89-L101)

**Comment by cvanderaa** [comment]
Note to self: 
* throw a message when at least one median CV is NA. This happens if a sample doesn't have any proteins that is mapped to more than 5 peptides
* Set the number of required peptides (here >5) as an argument

**Comment by cvanderaa** [comment]
The https://github.com/UCLouvain-CBIO/scp/commit/bc3b1071c95996da512329d1feb4d2aaaa482a88 commit refactors the function `computeMedianCV`. I renamed it to `computeMedianCV_SCoPE2`. I consider this function as deprecated but keep it in the package for the moment for backward compatibility with the SCoPE2 analysis replication. 

I created a new function `medianCVperCell` that should do more or less the same as `computeMedianCV_SCoPE2` but in a more standardized way, with matrix operations instead of `tidyverse` functions. Some additional changes to the function:
- The normalization that was previously convoluted into the function is now optional and is relying on `normalizeSCP` for more flexibility
- The CV are stored in the `colData` of the `QFeatures` object instead of the `SingleCellExperiment`. Furthermore, the user is allowed to choose the name of the new variable in the `colData`
- The user can put a threshold on the number of observations required to compute the CVs
- The function can be applied to multiple assays as long as they don't contain duplicated column names
- If the median CV is NA for one of the columns, a warning is thrown

This new function is based on an internal function [`featureCV`](https://github.com/UCLouvain-CBIO/scp/blob/bc3b1071c95996da512329d1feb4d2aaaa482a88/R/compute_metrics.R#L330-L351) that is inspired from the `MSnbase` implementation. The function takes a `SingleCellExperiment` object and a grouping variable, performs optionally a normalization procedure, computes CV and return a matrix with CVs. The function is not exported, but maybe this could be useful for example to use the CV to also filter feature.
